### PR TITLE
net: prerequisites for p2p encapsulation changes

### PR DESCRIPTION
**Pull request by theuni**
A few boring changes that are required before beginning to encapsulate the p2p code. For reference, https://github.com/theuni/bitcoin/tree/net-refactor12 builds on top of these as well as #7868. The high-level purpose of this work is to encapsulate p2p functionality into a single class rather than scattered globals, so that It may be safely changed without worrying about side-effects.

Regardless of that, I think these are all improvements in their own right. See individual commit messages for more info.

### CODE REVIEW

**Comment by sipa** [comment]
Is SetBannedIsDirty still needed after this?

**Comment by sipa** [comment]
utACK b33a17803a067d98c48f22b95eb4a5297eb03944

**Comment by btcdrak** [comment]
Concept ACK

File: src/net.cpp [hunk 1]
```diff
@@ -644,7 +679,7 @@ void CNode::copyStats(CNodeStats &stats)
     stats.dPingMin  = (((double)nMinPingUsecTime) / 1e6);
     stats.dPingWait = (((double)nPingUsecWait) / 1e6);
 
-    // Leave string empty if addrLocal invalid (not filled in yet)
+    // Leave std::string empty if addrLocal invalid (not filled in yet)
```
Comment (MarcoFalke) [inline src/net.cpp]: No need to change this as well, but it doesn't hurt much, either.

**Comment by MarcoFalke** [comment]
Concept ACK

File: src/net.cpp (hunk 1 above)
Comment (theuni) [inline src/net.cpp]: Heh, overzealous sed. Thanks, will fix.

**Comment by theuni** [comment]
@sipa The flag is still needed for non-user-initiated bans. As implemented here, Ban() doesn't flush to disk automatically for p2p violations.

**Comment by theuni** [comment]
Added a quick fixup for the string comment rather than rebasing and invalidating ACKs. I can squash before merge.

**Comment by sipa** [comment]
utACK 8d18913269e512b122e835b44ce1c3296a204177

**Comment by laanwj** [comment]
Needs rebase after #7868

**Comment by theuni** [comment]
Rebased

File: src/qt/rpcconsole.cpp [hunk 2]
```diff
@@ -892,8 +892,6 @@ void RPCConsole::banSelectedNode(int bantime)
         SplitHostPort(nStr, port, addr);
 
         CNode::Ban(CNetAddr(addr), BanReasonManuallyAdded, bantime);
-        bannedNode->fDisconnect = true;
```
Comment (paveljanik) [inline src/qt/rpcconsole.cpp]: The variable `bannedNode` seems to be unused here now.

The similar code is a few lines above in `RPCConsole::disconnectSelectedNode()` - same changes needed?

**Comment by paveljanik** [comment]
Concept ACK

**Comment by laanwj** [comment]
utACK 8fe811b (please get rid of the dummy commit)

**Comment by laanwj** [comment]
Needs rebase

**Comment by sipa** [comment]
@theuni Can you address https://github.com/bitcoin/bitcoin/pull/7906/commits/d48445a32faa50ff028bc58add54e67b33528c25#r60804931 ?

**Comment by theuni** [comment]
@sipa ah sorry, I missed that note. I'll have a look Monday.

File: src/qt/rpcconsole.cpp (hunk 2 above)
Comment (theuni) [inline src/qt/rpcconsole.cpp]: @paveljanik Yea, bannedNode is now unused here. I'll remove.

I only addressed banning here, so no changes needed for disconnectSelectedNode. Disconnect handling comes as a next step.

**Comment by sipa** [comment]
reutACK 5d5e7a097a87e0fe6efb5e2d622daadc10c2ad79

**Pull request merged**